python-client/src/aws_iatk/jsonrpc.py

```diff
@@ -30,7 +30,7 @@ def __init__(self, method: str, params: dict, region: str = None, profile: str =
         if profile:
             self.params["Profile"] = profile
 
-    def to_dict(self, caller: Optional[str]=None):
+    def to_dict(self, caller: dict):
```

Comment:
Should this not still be an Optional type?

Reply:
I don't think its needed because to_dict and dump_bytes are only ever called within the invoke_iatk function and caller is always defined there because there is a default value `def _invoke_iatk(self, payload: Payload, caller: dict = None) -> dict:`. So therefore adding the optional type doesnt change the logic of the code at all because it already exists